pkg/vfs/vfs.go

```diff
@@ -48,6 +50,63 @@ type Port struct {
 	PyroscopeAddr   string `json:",omitempty"`
 }
 
+// FuseOptions contains options for fuse mount, keep the same structure with `fuse.MountOptions`
+type FuseOptions struct {
+	AllowOther               bool
+	Options                  []string
+	MaxBackground            int
+	MaxWrite                 int
+	MaxReadAhead             int
+	MaxPages                 int
+	IgnoreSecurityLabels     bool // ignoring labels should be provided as a fusermount mount option.
+	RememberInodes           bool
+	FsName                   string
+	Name                     string
+	SingleThreaded           bool
+	DisableXAttrs            bool
+	Debug                    bool
+	EnableLocks              bool
+	ExplicitDataCacheControl bool
+	DirectMount              bool
+	DirectMountFlags         uintptr
+	EnableAcl                bool
+	EnableWriteback          bool
+	EnableIoctl              bool
+	DontUmask                bool
+	OtherCaps                uint32
+	NoAllocForRead           bool
+}
+
+func (o FuseOptions) StripOptions() FuseOptions {
+	options := o.Options
+	o.Options = make([]string, 0, len(o.Options))
+	for _, opt := range options {
+		if opt == "nonempty" {
+			continue
+		}
+		o.Options = append(o.Options, opt)
+	}
+
+	sort.Strings(o.Options)
+
+	// ignore these options because they won't be send to kernel
+	o.IgnoreSecurityLabels,
+		o.RememberInodes,
+		o.SingleThreaded,
+		o.DisableXAttrs,
```

Comment:
Once xattr is disabled, ENOSYS will be returned to kernel, then no more request will be sent to FUSE